packages/flame/lib/src/components/core/component.dart

```diff
@@ -206,6 +207,11 @@ class Component {
   void _setRemovingBit() => _state |= _removing;
   void _clearRemovingBit() => _state &= ~_removing;
 
+  /// Whether the component is removed.
```

Comment:
```suggestion
  /// Whether the component has been removed.
```
Since this will be false before it has been added to a parent.